app/validators/proxy_features_validator.rb

```diff
@@ -0,0 +1,15 @@
+class ProxyFeaturesValidator < ActiveModel::EachValidator
+  def initialize(args)
+    @features = args
+    @features.delete(:class)
+    super
+  end
+
+  def validate_each(record, attribute, value)
+    if value && !value.has_feature?(@features[attribute])
+      translated = _(@features[attribute])
```

Comment:
These strings don't seem to be extracted anywhere.  It'd be slightly preferable to extract them with the whole feature string in the models, e.g. `... does not have the DNS feature` as it lets translators use better grammar.